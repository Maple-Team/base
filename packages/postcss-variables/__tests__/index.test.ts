import postcss from 'postcss'
import plugin from '../src'

describe('postcss-variables test cases', () => {
  it('should pass with matching css', async () => {
    const result = postcss([
      plugin({
        variables: {
          ph55: `'ph55','other font family'`,
          ph65: `'ph65','other font family'`,
          ph75: `'ph75','other font family'`,
          textOpacity85: 'rgba(0,0,0,.85)',
          textOpacity65: 'rgba(0,0,0,.65)',
        },
      }),
    ])
      .process(
        `
        a{
          font-family: '$ph55';
          color: '$textOpacity85';
        }
        b{
          font-family: '$ph65';
          color: '$textOpacity65';
        }
        i{
          font-family: '$ph75';
        }
      `
      )
      .root.toResult().css
    expect(result).toMatchSnapshot()
  })
})
