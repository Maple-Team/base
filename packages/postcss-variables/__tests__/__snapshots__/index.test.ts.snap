// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss-variables test cases should pass with matching css 1`] = `
"
        a{
          font-family: 'ph55','other font family';
          color: rgba(0,0,0,.85);
        }
        b{
          font-family: 'ph65','other font family';
          color: rgba(0,0,0,.65);
        }
        i{
          font-family: 'ph75','other font family';
        }
      "
`;
